service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/database/AzureDatabaseUtilsRunner.java

```diff
@@ -388,7 +388,10 @@ private void runAzureDatabaseUtils(
       Map<String, String> secretStringData,
       String namespace)
       throws InterruptedException {
-    var aksApi = kubernetesClientProvider.createCoreApiClient(azureCloudContext, workspaceId);
+    var aksApi =
+        kubernetesClientProvider
+            .createCoreApiClient(azureCloudContext, workspaceId)
+            .orElseThrow(() -> new RuntimeException("No shared cluster found"));
```

Comment:
This is the same exception that was being thrown in the call, before - just moved to the call site instead.